templates/cinder/bin/healthcheck.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Trivial HTTP server to check health of scheduler, backup and volume services.
+# Cinder-API hast its own health check endpoint and does not need this.
+#
+# The only check this server currently does is using the heartbeat in the
+# database service table, accessing the DB directly here using the cinder.conf
+# configuration options.
+#
+# The benefit of accessing the DB directly is that it doesn't depend on the
+# Cinder-API service being up and we can also differentiate between the
+# container not having a connection to the DB and the cinder service not doing
+# the heartbeats.
+#
+# For volume services all enabled backends must be up to return 200, so it is
+# recommended to use a different pod for each backend to avoid one backend
+# affecting others.
+
+from http import server
+import sys
+
+from oslo_config import cfg
+
+from cinder import context
```

Comment:
Slight concern that importing openstack python modules in health checks could result in tight coupling of the operator version and the OSP release version. In the mid to long term it may be best to push this upstream

Reply:
Maybe upstream in the tripleo images?

Reply:
It could be relevant there, don't know enough about their criteria to be sure.